src/mx/NotificationSenderInterface.java

```diff
@@ -0,0 +1,9 @@
+package mx;
+
+/**
+ * Created by masterix on 28.05.2017.
+ */
+public interface NotificationSenderInterface {
+    void setDomain(Domain domain);
```

Comment:
Avoid setters in interfaces, unless absolutely necessary - interfaces should define contract (behavior), not tell anything about internal content of objects. Also, think about this - `NotificationSender` is something responsible for sending notification. Would you say setting a domain is natural part of sending notification?